src/app/services/gitUsers.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/observable/throw';
+import { ConfigService } from './config.service';
+
+@Injectable()
+export class GitUsersService {
+
+  constructor(private configService: ConfigService, private http: Http) {
+  }
+
+  public getUsers(searchText: string): Observable<SearchResult> {
+    // Since git API don't handle empty string
+    if (!searchText) {
+      return this.handleError('Invalid input');
+    }
```

Comment:
@TolyMelnikov  Agree .. more validation needed . Plz see above one of the validation. For this assessment, I am logging in console only but in past I developed browser logging which uses web-worker ( web-worker so that our precious single thread in browser is not used for logging, analytics ..etc).  BTW: I do understand that my above comment is not relevant to this PR.   But, I will plugin my old projects because I have to sell myself. :-) . I will send that logging code  by email because why not.